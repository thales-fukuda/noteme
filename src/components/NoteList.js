import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Menu from './blocks/Menu';

// Assets
import icMenu from '../assets/icons/menu.svg';
import icMinus from '../assets/icons/minus.svg';
import icPlus from '../assets/icons/plus.svg';

class NoteList extends Component {
  state = { menuOpen: false };

  toggleNav = () => {
    this.setState(prevState => ({
      menuOpen: !prevState.menuOpen,
    }));
  }

  render() {
    const { menuOpen } = this.state;
    const { notes, addNote, removeNote, activeNote } = this.props;
    return (
      <Menu bgColor='#F9CA24'>
        <Menu.Button icon={icMenu} onClick={this.toggleNav} />
        <Menu.Button icon={icPlus} onClick={addNote} />
        <Menu.Nav className={menuOpen ? 'active' : ''} bgColor='#E5E5E5'>
          {notes.map(el => (
            <Menu.Note key={el.id} onClick={() => activeNote(el.id)}>
              <Menu.Note.Title>{el.noteTitle}</Menu.Note.Title>
              <Menu.Button icon={icMinus} onClick={() => removeNote(el.id)} />
            </Menu.Note>
          ))}
        </Menu.Nav>
      </Menu>
    );
  }
}

NoteList.propTypes = {
  notes: PropTypes.instanceOf(Array).isRequired,
  addNote: PropTypes.func.isRequired,
  removeNote: PropTypes.func.isRequired,
  activeNote: PropTypes.func.isRequired,
};

export default NoteList;
