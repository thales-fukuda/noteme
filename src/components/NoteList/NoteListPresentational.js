import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Menu from '../blocks/Menu';

// Assets
import icMenu from '../../assets/icons/menu.svg';
import icMinus from '../../assets/icons/minus.svg';
import icPlus from '../../assets/icons/plus.svg';

class NoteListPresentational extends Component {
  state = { menuOpen: false };

  toggleNav = () => {
    this.setState(prevState => ({
      menuOpen: !prevState.menuOpen,
    }));
  };

  handleActiveNote = (e, id) => {
    e.stopPropagation();
    const { activeNote } = this.props;
    activeNote(id);
  };

  handleRemoveNote = (e, id) => {
    e.stopPropagation();
    const { removeNote } = this.props;
    removeNote(id);
  };

  render() {
    const { menuOpen } = this.state;
    const {
      notes,
      addNote,
    } = this.props;
    return (
      <Menu bgColor='#F9CA24'>
        <Menu.Button icon={icMenu} onClick={this.toggleNav} />
        <Menu.Button icon={icPlus} onClick={addNote} />
        <Menu.Nav className={menuOpen ? 'active' : ''} bgColor='#E5E5E5'>
          {notes.map(el => (
            <Menu.Note key={el.id} onClick={e => this.handleActiveNote(e, el.id)}>
              <Menu.Note.Title>{el.noteTitle}</Menu.Note.Title>
              <Menu.Button icon={icMinus} onClick={e => this.handleRemoveNote(e, el.id)} />
            </Menu.Note>
          ))}
        </Menu.Nav>
      </Menu>
    );
  }
}

NoteListPresentational.propTypes = {
  notes: PropTypes.instanceOf(Array).isRequired,
  addNote: PropTypes.func.isRequired,
  removeNote: PropTypes.func.isRequired,
  activeNote: PropTypes.func.isRequired,
};

export default NoteListPresentational;
