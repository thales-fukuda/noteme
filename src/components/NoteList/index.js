import NoteListWrapper from './NoteListWrapper';

export default NoteListWrapper;
