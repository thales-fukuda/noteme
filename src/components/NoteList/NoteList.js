import { connect } from 'react-redux';
import { addNote, removeNote, activeNote } from '../../redux/actions';
import NoteListPresentational from './NoteListPresentational';

const mapStateToProps = state => ({
  notes: state.notes.notes,
});

const mapDispatchToProps = dispatch => ({
  addNote: () => dispatch(addNote()),
  removeNote: id => dispatch(removeNote(id)),
  activeNote: id => dispatch(activeNote(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NoteListPresentational);
