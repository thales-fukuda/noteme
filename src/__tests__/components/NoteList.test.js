/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NoteList from '../../components/NoteList/NoteList';

window.matchMedia = jest.fn(query => ({ 
  matches: query.indexOf('(min-width: 726px)') !== -1, 
})); 

describe('<NoteList/>', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const fakeProps = {
        notes: [],
        activeNote: null,
        addNote: () => {},
        removeNote: () => {},
        updateActiveNote: () => {}
      }
      const wrapper = shallow(<NoteList {...fakeProps}/>).dive();
      expect(toJson(wrapper)).toMatchSnapshot;
    });
  });
});
