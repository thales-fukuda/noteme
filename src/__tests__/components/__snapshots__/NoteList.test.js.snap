// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteList/> render() renders the component 1`] = `
<StyledComponent
  bgColor="#F9CA24"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Button": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  padding: 0;
  height: 32px;
  width: 32px;
  background-image: url(",
            [Function],
            ");
  background-color: transparent;
  border: none;
",
          ],
        },
        "defaultProps": Object {
          "icon": "",
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "icon": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "CloseMenu": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  position: fixed;
  top: 70px;
  left: 0;
  height: calc(100vh - 70px);
  width: 100vw;
  background-color: rgba(0, 0, 0, .6);
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Nav": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "
  height: calc(100vh - 70px);
  width: 75%;
  position: fixed;
  top: 70px;
  left: 0;
  background-color: ",
            [Function],
            ";
  transform: translate3d(-100%, 0, 0);
  transition: ease .3s transform;
  overflow: scroll;
  z-index: 3;
  
  &.active{
    transform: translate3d(0, 0, 0);
  }

  @media (min-width: 726px) {
    width: 40%;
    transform: translate3d(0, 0, 0);
    box-shadow: 3px 0 6px rgba(0, 0, 0, .16);
  }
",
          ],
        },
        "defaultProps": Object {
          "bgColor": "#F2F2F2",
        },
        "displayName": "styled.nav",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "bgColor": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "nav",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Note": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Title": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  font-family: 'Lato';
  height: 40px;
  width: 100%;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
",
            ],
          },
          "displayName": "styled.h2",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "h2",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "rules": Array [
            "
  padding: 25px 15px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid #CECECE;
  font-weight: bold;
",
          ],
        },
        "defaultProps": Object {
          "active": ".5",
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "active": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "TitleWrapper": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": false,
          "rules": Array [
            "
  width: 100%;
  height: 100%;
  
  h2 {
    opacity: ",
            [Function],
            "
  }

  p {
    font-family: 'Lato';
    font-size: 13px;
    font-weight: 200;
    color: #bcbcbc;
  }
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  height: 70px;
  position: fixed;
  left: 0;
  top: 0;
  padding: 0 15px;
  box-shadow: 0 3px 6px rgba(0, 0, 0, .16);
  background-color: ",
          [Function],
          ";
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;

  @media (min-width: 726px) {
    box-shadow: 3px 0 6px rgba(0, 0, 0, .16);
    width: 40%;
  }
",
        ],
      },
      "defaultProps": Object {
        "bgColor": "#F9CA24",
      },
      "displayName": "styled.header",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bgColor": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "header",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.button
    icon="plus.svg"
    onClick={[Function]}
  />
  <styled.nav
    bgColor="#F2F2F2"
    className=""
  />
</StyledComponent>
`;
